modules/backend/classes/MainMenuItem.php

```diff
@@ -0,0 +1,125 @@
+<?php namespace Backend\Classes;
+
+use October\Rain\Exception\SystemException;
+
+/**
+ * Class MainMenuItem
+ *
+ * @package Backend\Classes
+ */
+class MainMenuItem
+{
+    /**
+     * @var string
+     */
+    public $code;
+
+    /**
+     * @var string
+     */
+    public $owner;
+
+    /**
+     * @var string
+     */
+    public $label;
+
+    /**
+     * @var null|string
+     */
+    public $icon;
+
+    /**
+     * @var null|string
+     */
+    public $iconSvg;
+
+    /**
+     * @var mixed
+     */
+    public $counter;
+
+    /**
+     * @var null|string
+     */
+    public $counterLabel;
+
+    /**
+     * @var string
+     */
+    public $url;
+
+    /**
+     * @var array
+     */
+    public $permissions = [];
+
+    /**
+     * @var int
+     */
+    public $order = 500;
+
+    /**
+     * @var SideMenuItem[]
+     */
+    public $sideMenu = [];
+
+    /**
+     * @param string $permission
+     * @param array $definition
+     */
+    public function addPermission(string $permission, array $definition)
+    {
+        $this->permissions[$permission] = $definition;
+    }
+
+    /**
+     * @param SideMenuItem $sideMenu
+     */
+    public function addSideMenuItem(SideMenuItem $sideMenu)
+    {
+        $this->sideMenu[$sideMenu->code] = $sideMenu;
+    }
+
+    /**
+     * @param string $code
+     * @return SideMenuItem
+     * @throws SystemException
+     */
+    public function getSideMenuItem(string $code): SideMenuItem
+    {
+        if (!array_key_exists($code, $this->sideMenu)) {
+            throw new SystemException('No sidenavigation item available with code ' . $code);
+        }
+
+        return $this->sideMenu[$code];
+    }
+
+    /**
+     * @param string $code
+     */
+    public function removeSideMenuItem(string $code)
+    {
+        unset($this->sideMenu[$code]);
+    }
+
+    /**
+     * @param array $data
+     * @return static
+     */
+    public static function createFromArray(array $data): self
```

Comment:
@Klaasie While I think `self` is technically valid here as a return type as this returns an instance of the same class, the correct return type is specifically `static`, however, that won't be coming until PHP 8. It's best to remove this for now. (The docblock is fine to keep though).

Refs:
- https://stackoverflow.com/questions/39068983/php-7-interfaces-return-type-hinting-and-self
- https://wiki.php.net/rfc/static_return_type
```suggestion
    public static function createFromArray(array $data)
```